### DISCUSSION TITLE
Why are boundaries are not 6 in total for all resolutions (5 for level 0 and 7 for level 1)?

### DISCUSSION THREAD

**Discussion by ysig** [uber/h3#755] [General]
Hi, 

I'm trying to compute the distances between a `lat,lon` point and its sequence of assigned hexagons in different resolutions.
To this end I have been using the following code:

```
def compute_distances(lat, lon, res):
  for r in range(0, 15)
    h = h3.latlng_to_cell(lat, lon, res)
    boundaries = h3.cell_to_boundary(h)
    distance_res = []
    for _, (lat_, lon_) in enumerate(boundaries):
        distance_res.append(abs(h3.great_circle_distance((lat, lon), (lat_, lon_), unit='km')))
```

what I've noticed however was that h3.cell_to_bondary doesn't return 6 hexagon points as expected but instead if I print:

```
>>> [len(h3.cell_to_boundary(h3.latlng_to_cell(0, 0, r))) for r in range(15)]
[5, 7, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6]
```
I see that the first two levels are 5 and 7. 

Could you please explain why and what does this mean?

Thanks,

**Comment by nrabinowitz** [comment]
There are two issues here, both related to the layout of the grid on top of the faces of an icosahedron (see [this blog post](https://www.uber.com/blog/h3/) for more details):
- It's not possible to tile a sphere with only hexagons. Every resolution of H3 will have exactly 12 pentagons in addition to the hexagon cells. This is because of the layout of cells on an icosahedron - each vertex of the icosahedron has 5 edges that come together, and must therefore have a pentagon shaped cell to cover it.
- Odd resolutions of H3 will have some cells with edges that cross the edges of the icosahedron, moving into a new coordinate system. While these cells are still topologically hexagons or pentagons - i.e. they have either 6 or 5 neighbors, respectively - we introduce "distortion vertexes" at each crossing point to merge the two coordinate systems.

Because of these issues, there are a number of cells at each resolution that may have more or less than 6 vertexes. At even resolutions, there are hexagons with 6 vertexes and pentagons with 5; at odd resolutions, there are hexagons with 6, 7, or 8 vertexes and pentagons with 10. See the ["Literal Edge Cases" section here](https://observablehq.com/@nrabinowitz/h3-index-inspector#cell-428) for examples of the different types of cell distortion.

**Reply by ysig** [reply]
Thank you so much for your always detailed answers! It's kind of unfortunate because this irregularity means that for my application I have to switch to S2 (or something else).

**Reply by nrabinowitz** [reply]
Yes, these are intractable issues with a hexagon grid - you can't tile the sphere with only hexagons, even if we came up with an alternate solution for the distortion vertexes. H3 isn't perfect, and there are certainly cases where square grids might be preferable.
